pkg/rke2/config.go

```diff
@@ -321,12 +321,10 @@ func newRKE2ServerConfig(opts ServerConfigOpts) (*ServerConfig, []bootstrapv1.Fi
 			})
 		}
 
-		rke2ServerConfig.EtcdSnapshotDir = opts.ServerConfig.Etcd.BackupConfig.S3.Folder
 		rke2ServerConfig.EtcdS3SkipSslVerify = !opts.ServerConfig.Etcd.BackupConfig.S3.EnforceSSLVerify
-	} else {
-		rke2ServerConfig.EtcdSnapshotDir = opts.ServerConfig.Etcd.BackupConfig.Directory
```

Comment:
@alexander-demicev @Danil-Grigorev thanks for reviews, PTAL once again. I updated the code so that `EtcdSnapshotDir` (local snapshot dir) set to `S3.Folder` logic is removed and it being set in else moved out of it since else condition is redundant here.